--- sddm-0.21.0/data/scripts/wayland-session~	2024-02-26 11:23:03.000000000 +0100
+++ sddm-0.21.0/data/scripts/wayland-session	2024-09-22 00:54:28.530781284 +0200
@@ -10,6 +10,9 @@
 # Backup the user shell setting into SDDM specific variable
 SDDM_USER_SHELL=$SHELL
 
+[ -f /etc/profile ] && . /etc/profile
+[ -f $HOME/.profile ] && . $HOME/.profile
+
 case $SHELL in
   */bash)
     [ -z "$BASH" ] && exec $SHELL --login $0 "$@"
@@ -38,8 +41,6 @@
     rm -f $xsess_tmp
     ;;
   *) # Plain sh, ksh, and anything we do not know.
-    [ -f /etc/profile ] && . /etc/profile
-    [ -f $HOME/.profile ] && . $HOME/.profile
     ;;
 esac
 
